### PR TITLE
kvrpc: add LockInfo & LeaderInfo into ResultType

### PR DESCRIPTION
**Pull request by morefreeze**
PTAL

### CODE REVIEW

File: proto/kvrpcpb.proto [hunk 1]
```diff
@@ -16,6 +16,16 @@ enum MessageType {
     CmdCommitThenGet = 7;
 }
 
+message LockInfo {
+    optional bytes primary_lock = 3;
```
Comment (siddontang) [inline proto/kvrpcpb.proto]: why starts from 3?

File: proto/kvrpcpb.proto [hunk 2]
```diff
@@ -24,11 +34,15 @@ message ResultType {
...
         Rolledback = 5;
+        NotLeader = 6;
         // Known result type add here.
         Other = 9;
     }
     optional Type type = 1 [default = Other];
     optional string msg = 2;
+    // If and only if Type == Locked
+    optional LockInfo lock_info = 3;
+    optional LeaderInfo leader_info = 4;
```
Comment (BusyJay) [inline proto/kvrpcpb.proto]: Why not use `NotLeader` in `errorpb.rs` directly?

File: proto/kvrpcpb.proto (hunk 1 above)
Comment (morefreeze) [inline proto/kvrpcpb.proto]: Copy mistake.

File: proto/kvrpcpb.proto (hunk 2 above)
Comment (morefreeze) [inline proto/kvrpcpb.proto]: It can use `NotLeaderError`, I don't like its type name.

**Comment by morefreeze** [comment]
@siddontang @BusyJay PTAL

**Comment by siddontang** [comment]
LGTM

**Comment by BusyJay** [comment]
LGTM

**Pull request merged**